app.rb

```diff
@@ -27,42 +30,6 @@ def list_of_people
     end
   end
 
-  def create_teacher
-    print "Teacher's Specialization: "
-    specialization = gets.chomp
-    print "Teacher's age: "
-    age = gets.chomp.to_i
-    print "Teacher's name: "
-    name = gets.chomp
-
-    teacher = Teacher.new(age, name, specialization, parent_permission: true)
-    @people << teacher
-    puts 'You have successfully registered a Teacher'
-  end
-
-  def create_student
-    print 'Age: '
-    age = gets.chomp.to_i
-    print 'Name: '
-    name = gets.chomp
-    print 'Has parent permission? [Y/N]: '
-
-    parent_permission = gets.chomp.downcase
-
-    case parent_permission
-    when 'n'
-      student = Student.new(nil, age, name, parent_permission: false)
-      @people << student
-    when 'y'
-      student = Student.new(nil, age, name, parent_permission: true)
-      @people << student
-    else
-      'You have entered an invalid option'
-    end
-
-    puts 'You have successfully registered a Student'
-  end
-
   def create_person
     print 'Do you want to create a new student (1) or teacher (2)? [input the number]: '
     selected_person_type = gets.chomp.to_i
```

Comment:
- [ ] @emhamza and @Jeddaa Great work refactoring your app, however, it's not good practice to have `create_person`,`create_book`, and   `create_rental` methods in your app.rb file, all methods that requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.

Reply:
@danmainah thank you for the feedback! we have moved all the inputs and inputs validations to the main.rb file. 💯

Reply:
@danmainah Thank you for your review but we are afraid that as per the project requirements the main.rb should only to start your code and ask the user for the initial choice of options (or to quit). Please have a look or elaborate your feedback clearly: 🔢 

![required](https://github.com/emhamza/opp-school-library/assets/122353030/d9ae3f31-f7f5-446b-890f-8dae19925583)